MariaDB/server
# MDEV-31045: Update debian/rules to be in sync to MariaDB LTS 10.6

- [x] *The Jira issue number for this PR is: MDEV-31045*


## Description
To make sure that PMEM problem does not happen again sync 10.9 and up debian/rules to make sure that they are not making anymore problems with autobake-debs.sh

MariaDB 10.8 have correct changes but 10.9 and above does not have needed changes to make PR #2597 to work

## How can this PR be tested?
This can tested to run on Debian machine `debian/autobake-debs.sh` and see if it works.

## Basing the PR against the correct MariaDB version


## Backward compatibility
This ensures more backward compatibility

## PR quality check
